src/v2/api/services/ScheduleService.ts

```diff
@@ -49,17 +50,40 @@ export class ScheduleService {
   getIndexOfLesson (period, startTime, endOfWeek) {
     const difference = endOfWeek.getTime() - startTime.getTime();
     let index = difference / (1000 * 60 * 60 * 24 * 7);
-    if (period === Period.EVERY_FORTNIGHT) index /= 2;
 
-    if (index % 1 <= 0.5) {
+    if (period === Period.EVERY_WEEK) {
       return +parseInt(String(index));
-    } else {
-      return null;
+    } else if (period === Period.EVERY_FORTNIGHT) {
+      index /= 2;
+      if (index % 1 <= 0.5) {
+        return +parseInt(String(index));
+      } else {
+        return null;
+      }
+    } else if (period === Period.NO_PERIOD) {
+      return 0;
     }
   }
 
-  async getGeneralGroupEvents (id, week) {
+  private isWeekValid (week: number, currentSemester: CurrentSemester): boolean {
+    const startWeek = Math.ceil(
+      (currentSemester.startDate.getTime() - currentSemester.endDate.getTime()) / WEEK
+    );
+    const endWeek = Math.ceil(
+      (currentSemester.endDate.getTime() - currentSemester.startDate.getTime()) / WEEK
+    );
+
+    return week >= startWeek && week <= endWeek;
+  }
+
+  async getGeneralGroupEvents (id: string, week: number) {
+    const currentSemester = await this.dateService.getCurrentSemester();
+    const weekToCheck = week || (!currentSemester.isFinished ? await this.dateService.getCurrentWeek() : null);
+    if (weekToCheck === null || !this.isWeekValid(weekToCheck, currentSemester)) {
+      throw new InvalidWeekException();
+    }
     const { startOfWeek, endOfWeek } = week ? await this.dateService.getDatesOfWeek(week) : this.dateService.getDatesOfCurrentWeek();
+
```

Comment:
please, add in findMany field period, and that field can not be NO_PERIOD

Reply:
resolved in https://github.com/fictadvisor/fictadvisor-api/pull/564